------------------------------------------------------------
revno: 887 [merge]
fixes bug: https://launchpad.net/bugs/1236594
committer: Scott Moser <smoser@ubuntu.com>
branch nick: trunk
timestamp: Mon 2013-10-07 20:08:13 -0400
message:
  fix bug mounting first partition of a alias'd name.
  
  The expected behavior was that 'ephemeral0' in a mount device entry
  and ephemeral0 mapped to /dev/xvdb that /dev/xvdb1 or /dev/xvdb would
  be substituted.
  
  Explicitly setting 'ephemeral0.0' would mean only xvdb in this case.
------------------------------------------------------------
Use --include-merged or -n0 to see merged revisions.
=== modified file 'ChangeLog'

=== modified file 'cloudinit/config/cc_mounts.py'
--- a/cloudinit/config/cc_mounts.py
+++ b/cloudinit/config/cc_mounts.py
@@ -222,9 +222,6 @@ def devnode_for_dev_part(device, partiti
     if not os.path.exists(device):
         return None
 
-    if not partition:
-        return device
-
     short_name = os.path.basename(device)
     sys_path = "/sys/block/%s" % short_name
 
@@ -238,8 +235,7 @@ def devnode_for_dev_part(device, partiti
         partition = str(partition)
 
     if partition is None:
-        valid_mappings = [sys_long_path + "1",
-                          sys_long_path + "p1" % partition]
+        valid_mappings = [sys_long_path + "1", sys_long_path + "p1"]
     elif partition != "0":
         valid_mappings = [sys_long_path + "%s" % partition,
                           sys_long_path + "p%s" % partition]
